### PR TITLE
+str Add interleaveAll stream operator.

### PR DESCRIPTION
**Pull request by He-Pin**
References https://github.com/akka/akka/issues/19045

### CODE REVIEW

**Comment by johanandren** [comment]
(Close-reopen to retrigger PR validation)

File: akka-stream/src/main/scala/akka/stream/javadsl/Source.scala [hunk 1]
```diff
@@ -1609,6 +1609,34 @@ final class Source[Out, Mat](delegate: scaladsl.Source[Out, Mat]) extends Graph[
...
+   * '''Completes when''' the [[Flow]] and given [[Source]] completes
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def interleaveAll(
+      those: java.util.List[Source[Out, _ <: Any]],
+      segmentSize: Int,
+      eagerClose: Boolean): javadsl.Source[Out, Mat] = {
+    val seq = if (those != null) Util.immutableSeq(those).map(_.asScala) else immutable.Seq()
+    new Source(delegate.interleaveAll(seq, segmentSize, eagerClose))
```
Comment (He-Pin) [inline akka-stream/src/main/scala/akka/stream/javadsl/Source.scala:1637]: `those: java.util.List[_<:Graph[SourceShape[Out], _<:Any]]` ?

File: akka-stream/src/main/scala/akka/stream/javadsl/Source.scala [hunk 2]
```diff
@@ -1609,6 +1609,34 @@ final class Source[Out, Mat](delegate: scaladsl.Source[Out, Mat]) extends Graph[
...
+  }
```
Comment (He-Pin) [inline akka-stream/src/main/scala/akka/stream/javadsl/Source.scala:1638]:
```scala
  def interleaveAll(
      those: java.util.List[_<:Graph[SourceShape[Out], _<:Any]],
      segmentSize: Int,
      eagerClose: Boolean): javadsl.Source[Out, Mat] = {
    val seq = if (those != null) Util.immutableSeq(those).collect{
      case source:Source[Out @uncheckedVariance,_] => source.asScala
      case other => other
    } else immutable.Seq()
    new Source(delegate.interleaveAll(seq, segmentSize, eagerClose))
  }
```

**Comment by He-Pin** [comment]
This one is ready for review.

**Review by johanandren — APPROVED** [review]
LGTM!

**Comment by He-Pin** [comment]
@patriknw I think this one is ready for review.

File: akka-stream/src/main/scala/akka/stream/javadsl/Source.scala [hunk 3]
```diff
@@ -1609,6 +1609,37 @@ final class Source[Out, Mat](delegate: scaladsl.Source[Out, Mat]) extends Graph[
       matF: function.Function2[Mat, M, M2]): javadsl.Source[Out, M2] =
     new Source(delegate.interleaveMat(that, segmentSize, eagerClose)(combinerToScala(matF)))
 
+  /**
+   * Interleave is a deterministic merge of the given [[Source]] with elements of this [[Flow]].
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/javadsl/Source.scala:1613]: This is in `Source` so there is no `Flow`?

**Review by patriknw — COMMENTED** [review]
looking good, just something in the api docs

File: akka-stream/src/main/scala/akka/stream/javadsl/Source.scala (hunk 3 above)
Comment (patriknw) [reply akka-stream/src/main/scala/akka/stream/javadsl/Source.scala:1613]: but I see we have that for other things as well in javadsl.Source. In scaladsl the same FlowOps is used.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**